### DISCUSSION TITLE
Can files be handled with Tauri commands?

### DISCUSSION THREAD

**Discussion by ukihot** [tauri-apps/tauri#9566] [Q&A] [answered]
I want to pass a PDF file received on the frontend to backend. The command I'm trying to execute looks like this:

```rust
use std::fs::File;

#[tauri::command]
pub async fn handle_pdf(file: File) -> Result<&str, String> {
    println!("{:?}", file);
    Ok("")
}
```

I anticipated that this wouldn't work straight away. Leaving aside the internal processing, I'm unsure about the type to use when receiving it as an argument... Currently, I'm encountering the following error:

```log
the trait bound `std::fs::File: tauri::command::CommandArg<'_, tauri_runtime_wry::Wry<tauri::EventLoopMessage>>` is not satisfied
the following other types implement trait `tauri::command::CommandArg<'de, R>`:
  tauri::AppHandle<R>
  tauri::State<'r, T>
  tauri::Window<R>
required for `std::fs::File` to implement `tauri::command::CommandArg<'_, tauri_runtime_wry::Wry<tauri::EventLoopMessage>>`
```

**Comment by FabianLars** [comment, marked as answer]
std::fs::File has no javascript or json representation so using it as a command arugments can't work.
A similar problem on the other side: the javascript File also does not have a json representation so it can't be send directly. The only real solution here that i'm aware of is to send the file as plain binary.

The argument on the rust side would be `Vec<u8>` and on js ideally number[] but iirc the invoke api will convert Uint8arrays and arraybuffer to number[] for you (but it has to be number[] in the end for the ipc to work)

**Reply by ukihot** [reply]
Understood. Thank you very much for the swift reply.
